### PR TITLE
Small indexing fixes

### PR DESCRIPTION
**Pull request by schmidt-sebastian**

### CODE REVIEW

File: firebase-firestore/src/main/java/com/google/firebase/firestore/FirebaseFirestore.java [hunk 1]
```diff
@@ -307,6 +307,10 @@ public FirebaseApp getApp() {
   Task<Void> setIndexConfiguration(String json) {
     ensureClientConfigured();
 
+    if (!settings.isPersistenceEnabled()) {
+      throw new IllegalStateException("Cannot enable indexes when persisted is disabled");
```
Comment (thebrianchen) [inline firebase-firestore/src/main/java/com/google/firebase/firestore/FirebaseFirestore.java:311]: s/persisted/persistence

File: firebase-firestore/src/main/java/com/google/firebase/firestore/FirebaseFirestore.java [hunk 2]
```diff
@@ -320,9 +324,9 @@ public FirebaseApp getApp() {
       JSONObject jsonObject = new JSONObject(json);
 
       if (jsonObject.has("indexes")) {
-        JSONArray indices = jsonObject.getJSONArray("indexes");
-        for (int i = 0; i < indices.length(); ++i) {
-          JSONObject definition = indices.getJSONObject(i);
+        JSONArray indexes = jsonObject.getJSONArray("indexes");
```
Comment (thebrianchen) [inline firebase-firestore/src/main/java/com/google/firebase/firestore/FirebaseFirestore.java:327]: nit: rename `parsedIndices` to `parsedIndexes`.

**Review by thebrianchen — APPROVED** [review]
lgtm

File: firebase-firestore/src/main/java/com/google/firebase/firestore/FirebaseFirestore.java (hunk 1 above)
Comment (schmidt-sebastian) [reply firebase-firestore/src/main/java/com/google/firebase/firestore/FirebaseFirestore.java:311]: Done

**Pull request merged**